superset/assets/javascripts/explorev2/components/SaveModal.js

```diff
@@ -0,0 +1,249 @@
+    this.setState({ action });
+  }
+  changeDash(dash) {
+    this.setState({ addToDash: dash });
+  }
+  saveOrOverwrite(gotodash) {
+    this.setState({ alert: null });
+    this.props.actions.removeSaveModalAlert();
+    const params = {};
+    const sliceParams = {};
+    params.datasource_id = this.props.form_data.datasource;
+    params.datasource_type = this.props.datasource_type;
+    params.datasource_name = this.props.form_data.datasource_name;
+
+    let sliceName = null;
+    sliceParams.action = this.state.action;
+    if (sliceParams.action === 'saveas') {
+      sliceName = this.state.newSliceName;
+      if (sliceName === '') {
+        this.setState({ alert: 'Please enter a slice name' });
+        return;
+      }
+      sliceParams.slice_name = sliceName;
+    } else {
+      sliceParams.slice_name = this.props.form_data.slice_name;
+    }
+
+    Object.keys(this.props.form_data).forEach((field) => {
+      if (this.props.form_data[field] !== null && field !== 'slice_name') {
+        params[field] = this.props.form_data[field];
+      }
+    });
+
+    const addToDash = this.state.addToDash;
+    sliceParams.add_to_dash = addToDash;
+    let dashboard = null;
+    switch (addToDash) {
+      case ('existing'):
+        dashboard = this.state.saveToDashboardId;
+        if (!dashboard) {
+          this.setState({ alert: 'Please select a dashboard' });
+          return;
+        }
+        sliceParams.save_to_dashboard_id = dashboard;
+        break;
+      case ('new'):
+        dashboard = this.state.newDashboardName;
+        if (dashboard === '') {
+          this.setState({ alert: 'Please enter a dashboard name' });
+          return;
+        }
+        sliceParams.new_dashboard_name = dashboard;
+        break;
+      default:
+        dashboard = null;
+    }
+    params.V2 = true;
+    sliceParams.goto_dash = gotodash;
+    const baseUrl = '/superset/explore/' +
+      `${this.props.datasource_type}/${this.props.form_data.datasource}/`;
+    const saveUrl = `${baseUrl}?${$.param(params, true)}&${$.param(sliceParams, true)}`;
+    this.props.actions.saveSlice(saveUrl);
+    this.props.onHide();
+  }
+  removeAlert() {
+    if (this.props.alert) {
+      this.props.actions.removeSaveModalAlert();
+    }
+    this.setState({ alert: null });
+  }
+  render() {
+    return (
+      <Modal
+        show
+        onHide={this.props.onHide}
+        bsStyle="large"
+      >
+        <Modal.Header closeButton>
+          <Modal.Title>
+            Save A Slice
+          </Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          {(this.state.alert || this.props.alert) &&
+            <Alert>
+              {this.state.alert ? this.state.alert : this.props.alert}
+              <i
+                className="fa fa-close pull-right"
+                onClick={this.removeAlert.bind(this)}
+                style={{ cursor: 'pointer' }}
+              />
+            </Alert>
+          }
+          <form>
```

Comment:
would be great to use the react-bootstrap form components here: https://react-bootstrap.github.io/components.html#forms